container/container.go

```diff
@@ -162,11 +167,17 @@ func (container *Container) FromDisk() error {
 		return err
 	}
 
-	// Ensure the operating system is set if blank. Assume it is the OS of the
-	// host OS if not, to ensure containers created before multiple-OS
-	// support are migrated
+	// Ensure the platform is set. Assume it is the host platform otherwise
+	// to ensure containers created before multiple-OS support are migrated.
 	if container.OS == "" {
-		container.OS = runtime.GOOS
+		container.ImagePlatform = platforms.DefaultSpec()
+		container.OS = container.ImagePlatform.OS
+	} else {
+		// OS was deprecated in favor of ImagePlatform
+		// Make sure we migrate the OS to ImagePlatform.OS.
+		if container.ImagePlatform.OS == "" {
+			container.ImagePlatform.OS = container.OS
```

Comment:
In practice (for now) it won't matter much because we're mainly using the `ImagePlatform.OS` in other places, but for migrating containers I do worry about us not populating other fields here (like `ImagePlatform.Architecture` and it being annoying later.

Reply:
Done, the only case where it wouldn't be able to migrate the platform correctly is the case where the container image is no longer available. In such case it will fallback to the host platform.

Reply:
OTOH.. Not sure about the fallback. Maybe in this case, it's better for the `Architecture` to be empty, rather than assuming a potentially wrong arch.

WDYT?

Reply:
Yeah, it might be better not to fallback in that case. Barring any special case, I'd rather it be clear that we weren't able to deduce the platform than to populate that with a mistaken value.

Reply:
Removed the fallback